inference-engine/src/inference_engine/network_serializer.cpp

```
@@ -4,165 +4,265 @@
+                                     const ngraph::op::Constant &c) {
+  ConstantAtributes attr;
+  const uint8_t *p = reinterpret_cast<const uint8_t *>(c.get_data_ptr());
+  attr.size = ngraph::shape_size(c.get_shape()) * c.get_element_type().size();
+  attr.offset = bin.size();
+  bin.insert(end(bin), p, p + attr.size);
+  return attr;
+}
+
+void ngfunction_2_irv10(pugi::xml_document &doc, std::vector<uint8_t> &bin,
+                        const ngraph::Function &f) {
+  pugi::xml_node netXml = doc.append_child("net");
+  netXml.append_attribute("name").set_value(f.get_friendly_name().c_str());
+  netXml.append_attribute("version").set_value("10");
+  pugi::xml_node layers = netXml.append_child("layers");
+
+  auto layer_ids = create_layer_ids(f);
+  for (auto node : f.get_ordered_ops()) {
+    // <layers>
+    pugi::xml_node layer = layers.append_child("layer");
+    layer.append_attribute("id").set_value(layer_ids[node.get()]);
+    layer.append_attribute("name").set_value(node->get_friendly_name().c_str());
+    layer.append_attribute("type").set_value(node->get_type_name());
+    // TODO: calculate real opset
+    layer.append_attribute("version").set_value("opset1");
 
-        if (execGraphInfoSerialization) {
-            pugi::xml_document doc;
-            FillXmlDocWithExecutionNGraph(network, doc);
+    // <layers/data>
+    pugi::xml_node data = layer.append_child("data");
+
+    // <layers/data> general atributes
+    XmlVisitor visitor{data};
+    try {
+      if (!node->visit_attributes(visitor)) {
+        #if 0 // TODO: need to decide if we want to throw on vistiation fail
+        THROW_IE_EXCEPTION << "cannot visit " << node->get_name();
+        #else
+        std::cout << "cannot visit " << node->get_name() << "\n";
+        #endif
+      }
+    } catch (std::logic_error &ex) {
+#if 0 // TODO: need to decide if we want to throw on vistiation fail
+       THROW_IE_EXCEPTION << ex.what() << "from " << node->get_name();
```

I think if attributes cannot be read due to some error, we can throw an exception.

---

Please rebase to master or merge master to your branch - to fetch current OpenVino ONNX CI state